### ISSUE TITLE
fractional_max_pool2d_with_indices silently ignores output_ratio if output_size is provided

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#24031]
Reading the code: https://github.com/pytorch/pytorch/blob/32efb431294d99a60899b0809c6363065608e556/torch/nn/functional.py#L339-L350 , it looks like if both output_ratio and output_size are provided, then the function ignores output_ratio. There should be a check for this.

**Comment by tczhangzhi** [comment]
Yes, what is your expected result?  If I want to have an output size as a ratio of the input size, I won't define `output_size` at the same time~
